File: endpoint/templates/go/endpoint.controller.go
package <%= nameLowerCase %>controller

import (
	_ "encoding/json"
	_ "github.com/<%= username %>/<%= appName %>/server/api/<%= feature %>/dao"
	_ <%= nameLowerCase %> "github.com/<%= username %>/<%= appName %>/server/api/<%= feature %>/model"
	"github.com/julienschmidt/httprouter"
	_ "io/ioutil"
	_ "net/http"
)

func GetAll(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
}

func GetById(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
}

func New(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
}

func Update(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
}

func Remove(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {

}

File: endpoint/templates/node/typescript/endpoint.controller.ts
"use strict";

import * as express from 'express';
import {<%= name %>DAO} from '../dao/<%= name %>-dao';

export class <%= name %>Controller {
  static getAll(req:express.Request, res:express.Response) {
    <%= name %>DAO
      ['getAll']()
      .then(<%= nameLowerCase %> => res.status(200).json(<%= nameLowerCase %>s))
      .catch(error => res.status(400).json(error));
  }

  static createNew(req:express.Request, res:express.Response) {
    let _<%= nameLowerCase %> = req.body;

    <%= name %>DAO
      ['createNew'](_<%= nameLowerCase %>)
      .then(<%= nameLowerCase %> => res.status(201).json(<%= nameLowerCase %>))
      .catch(error => res.status(400).json(error));
  }

  static remove(req:express.Request, res:express.Response) {
    let _id = req.params.id;

    <%= name %>DAO
      ['removeById'](_id)
      .then(() => res.status(200).end())
      .catch(error => res.status(400).json(error));
  }
}
